### PR TITLE
enabled all code in gammu.c

### PR DESCRIPTION
**Pull request by kleinsasserm**
Hi! Now I've enabled all code in gammu.c -  I've tested it and I'am still able to send sms. Maybe you can have a look at it. Sorry that I commited the setup.py once again....

### CODE REVIEW

File: gammu/src/gammu.c [hunk 1]
```diff
@@ -5505,23 +5539,24 @@ gammu_SetDebugFile(PyObject *self, PyObject *args, PyObject *kwds)
         }
         error = GSM_SetDebugFileDescriptor(NULL, FALSE, GSM_GetGlobalDebug());
         if (!checkError(NULL, error, "SetDebugFileDescriptor")) return NULL;
-    } else if (PyFile_Check(value)) {
+    } else if (streamFromPyFile(value)) {
```
Comment (nijel) [inline gammu/src/gammu.c]: This leaks file descriptor.

File: gammu/src/gammu.c [hunk 2]
```diff
@@ -5505,23 +5539,24 @@ gammu_SetDebugFile(PyObject *self, PyObject *args, PyObject *kwds)
...
             Py_DECREF(DebugFile);
             DebugFile = NULL;
         }
-        f = PyFile_AsFile(value);
+        f = streamFromPyFile(value);
         if (f == NULL) return NULL;
         DebugFile = value;
         Py_INCREF(DebugFile);
         error = GSM_SetDebugFileDescriptor(f, FALSE, GSM_GetGlobalDebug());
         if (!checkError(NULL, error, "SetDebugFileDescriptor")) return NULL;
```
Comment (nijel) [inline gammu/src/gammu.c]: As the file descriptor is newly allocated now, you should make it closable.

File: setup.py [hunk 3]
```diff
@@ -26,6 +26,8 @@
 from setuptools import setup, Extension
 import os
 
+os.environ["CFLAGS"] = '-g -O0 -fno-inline'
```
Comment (nijel) [inline setup.py]: What's reason for this?

File: gammu/src/errors.c [hunk 4]
```diff
@@ -37,6 +37,8 @@ int checkError(GSM_StateMachine *s, GSM_Error error, const char *where) {
         err = gammu_error_map[error];
 
     msg = GSM_ErrorString(error);
+        
+    PyErr_SetString(PyExc_IOError, msg);
```
Comment (nijel) [inline gammu/src/errors.c]: Should not this be inside the if below?

File: gammu/src/gammu.c [hunk 5]
```diff
@@ -449,7 +474,9 @@ StateMachine_SetConfig(StateMachineObject *self, PyObject *args, PyObject *kwds)
     }
 
     while (PyDict_Next(dict, &pos, &key, &value)) {
-        s = PyString_AsString(key);
+        PyObject *ts = PyUnicode_AsUTF8String(key);
```
Comment (nijel) [inline gammu/src/gammu.c]: This creates new object, you need to deallocate it later (Py_DECREF).

File: gammu/src/gammu.c [hunk 6]
```diff
@@ -486,7 +513,9 @@ StateMachine_SetConfig(StateMachineObject *self, PyObject *args, PyObject *kwds)
                     str = value;
                     Py_INCREF(str);
                 }
-                v = PyString_AsString(value);
+                PyObject *tv = PyUnicode_AsUTF8String(value);
```
Comment (nijel) [inline gammu/src/gammu.c]: Again, resource leak.

File: gammu/src/gammu.c [hunk 7]
```diff
@@ -501,7 +530,9 @@ StateMachine_SetConfig(StateMachineObject *self, PyObject *args, PyObject *kwds)
                 } else {
                     typeobj = PyObject_Type(value);
                     typestr = PyObject_Str(typeobj);
-                    PyErr_Format(PyExc_ValueError, "Non string value for %s: %s", s, PyString_AsString(typestr));
+                    PyObject *tstr = PyUnicode_AsUTF8String(typestr);
```
Comment (nijel) [inline gammu/src/gammu.c]: Resource leak.

File: gammu/src/gammu.c [hunk 8]
```diff
@@ -5038,16 +5069,19 @@ StateMachine_SetDebugFile(StateMachineObject *self, PyObject *args, PyObject *kw
...
         if (f == NULL) return NULL;
         self->DebugFile = value;
         Py_INCREF(value);
 
         error = GSM_SetDebugFileDescriptor(f, FALSE, di);
         if (!checkError(NULL, error, "SetDebugFileDescriptor")) return NULL;
-    } else if (PyString_Check(value)) {
-        s = PyString_AsString(value);
+    } else if (PyBytes_Check(value)) {
+        PyObject *ts = PyUnicode_AsUTF8String(value);
```
Comment (nijel) [inline gammu/src/gammu.c]: Using PyUnicode_* on PyBytes seems to be wrong.

File: setup.py (hunk 3 above)
Comment (kleinsasserm) [inline setup.py]: That was commited by mistake. It was needed for debugging with gdb.

Comment (nijel) [inline setup.py]: You can run ```CFLAGS=-Wextra ./setup.py build``` without need to modify the code.

Comment (kleinsasserm) [inline setup.py]: Good to know! :+1:  :)

**Comment by nijel** [comment]
BTW: I've picked up module and object initialization parts from your patch and merged them in a way it works for both Python 2 and 3.

**Comment by kleinsasserm** [comment]
Oh, cool!

**Comment by nijel** [comment]
I've merged most of the strings changes (and doing some differently), so the code now compiles. But it's still quite broken as many tests fail. Anyway I'm closing this pull request as it has nothing what would be mergeable.

**Pull request closed without merging**